model_analyzer/config/generate/neighborhood.py

```diff
@@ -278,32 +280,37 @@ def _get_step_vector(self) -> Coordinate:
             compare_constraints=compare_constraints)
 
     def _calculate_step_vector_from_measurements(
-            self, compare_constraints: bool) -> Coordinate:
+            self, compare_constraints: bool) -> List[float]:
 
         home_measurement = self._get_home_measurement()
+        if not home_measurement:
+            raise Exception("Can't step from home if it has no measurement") 
+
         vectors, measurements = self._get_all_measurements()
 
         # This function should only ever be called if all are passing or none are passing
         _, p = self._get_measurements_passing_constraints()
         assert (len(p) == 0 or len(p) == len(measurements))
 
         if not vectors:
-            return Coordinate([0] * self._config.get_num_dimensions())
+            return ([0.0] * self._config.get_num_dimensions())
 
         weights = []
         for m in measurements:
             if compare_constraints:
                 weight = home_measurement.compare_constraints(m)
             else:
                 weight = home_measurement.compare_measurements(m)
+            if not weight:
+                weight = 0.0
             weights.append(weight)
 
         return self._calculate_step_vector_from_vectors_and_weights(
             vectors, weights)
 
-    def _calculate_step_vector_from_vectors_and_weights(self, vectors, weights):
-        step_vector = Coordinate([0] * self._config.get_num_dimensions())
-        dim_sum_vector = Coordinate([0] * self._config.get_num_dimensions())
+    def _calculate_step_vector_from_vectors_and_weights(self, vectors: List[Coordinate], weights: List[float]) -> List[float]:
+        step_vector = [0.0] * self._config.get_num_dimensions()
```

Comment:
What is this doing? Is it creating N [0.0] entries?

Reply:
Creates an array of length (get_num_dimensions()) where each entry is the float 0.0